dependency_support/at_clifford_yosys/bundled.BUILD.bazel

```diff
@@ -0,0 +1,441 @@
+# limitations under the License.
+
+# Yosys is a framework for RTL synthesis tools. It currently has extensive
+# Verilog-2005 support and provides a basic set of synthesis algorithms for
+# various application domains.
+
+licenses(["restricted"])
+
+exports_files(["LICENSE"])
+
+load("@rules_hdl//dependency_support/at_clifford_yosys:yosys.bzl", "yosys_syntax_check")
+load("@rules_hdl//dependency_support/com_github_westes_flex:flex.bzl", "genlex")
+load("@rules_hdl//dependency_support/org_gnu_bison:bison.bzl", "genyacc")
+
+genrule(
+    name = "version_src",
+    outs = ["kernel/version.cc"],
+    cmd = "echo 'namespace Yosys { extern const char *yosys_version_str; const char *yosys_version_str=\"Yosys 0.8+\"; }' >$@",
+)
+
+cc_library(
+    name = "version",
+    srcs = ["kernel/version.cc"],
+)
+
+genrule(
+    name = "techmap",
+    srcs = [
+        "techlibs/common/techmap.v",
+    ],
+    outs = ["passes/techmap/techmap.inc"],
+    cmd = (
+        "echo '// autogenerated from techmap.v' > $@;" +
+        "echo 'static char stdcells_code[] = {' >> $@;" +
+        "od -v -td1 -An $< | sed -e 's/[0-9][0-9]*/&,/g' >> $@;" +
+        "echo '0};' >> $@"
+    ),
+)
+
+genlex(
+    name = "verilog_lexer",
+    src = "frontends/verilog/verilog_lexer.l",
+    out = "verilog_lexer.cc",
+)
+
+genyacc(
+    name = "verilog_parser",
+    src = "frontends/verilog/verilog_parser.y",
+    header_out = "frontends/verilog/verilog_parser.tab.hh",
+    source_out = "frontends/verilog/verilog_parser.tab.cc",
+)
+
+genlex(
+    name = "ilang_lexer",
+    src = "frontends/ilang/ilang_lexer.l",
+    out = "ilang_lexer.cc",
+)
+
+genyacc(
+    name = "ilang_parser",
+    src = "frontends/ilang/ilang_parser.y",
+    header_out = "frontends/ilang/ilang_parser.tab.hh",
+    source_out = "frontends/ilang/ilang_parser.tab.cc",
+)
+
+py_binary(
+    name = "cellhelp",
+    srcs = ["techlibs/common/cellhelp.py"],
+    python_version = "PY3",
+    srcs_version = "PY2AND3",
+    deps = [
+    ],
+)
+
+genrule(
+    name = "simlib_help",
+    srcs = [
+        "techlibs/common/simlib.v",
+    ],
+    outs = ["techlibs/common/simlib_help.inc"],
+    cmd = (
+        "$(location :cellhelp) $(SRCS) > $@"
+    ),
+    tools = [":cellhelp"],
+)
+
+genrule(
+    name = "simcells_help",
+    srcs = [
+        "techlibs/common/simcells.v",
+    ],
+    outs = ["techlibs/common/simcells_help.inc"],
+    cmd = (
+        "$(location :cellhelp) $(SRCS) > $@"
+    ),
+    tools = [":cellhelp"],
+)
+
+GENERATED_HEADERS = [
+    "passes/pmgen/ice40_dsp_pm.h",
+    "passes/pmgen/ice40_wrapcarry_pm.h",
+    "passes/pmgen/test_pmgen_pm.h",
+    "passes/pmgen/peepopt_pm.h",
+    "passes/pmgen/xilinx_dsp_cascade_pm.h",
+    "passes/pmgen/xilinx_dsp_CREG_pm.h",
+    "passes/pmgen/xilinx_dsp_pm.h",
+    "passes/pmgen/xilinx_dsp48a_pm.h",
+    "passes/pmgen/xilinx_srl_pm.h",
+]
+
+YOSYS_COPTS = [
+    "-DYOSYS_SRC='\"./\"'",
+    "-DYOSYS_DATDIR='\"./\"'",
```

Comment:
Can you change this to.

`"-DYOSYS_DATDIR='std::getenv(\"YOSYS_DATDIR\") ? std::getenv(\"YOSYS_DATDIR\") : \"third_party/yosys/share\"'"`

Otherwise it will cause problems when we try to use yosys in custom rules.

Reply:
Done